lib/clients/backbeat-2017-07-01.api.json

```diff
@@ -949,10 +949,64 @@
                 }
             }
         },
+        "GetRaftID": {
+            "http": {
+                "method": "GET",
+                "requestUri": "/_/metadata/admin/buckets/{Bucket}/id"
+            },
+            "input": {
+                "type": "string",
+                "required": [
+                    "Bucket"
+                ],
+                "members": {
+                    "Bucket": {
+                        "location": "uri",
+                        "locationName": "Bucket"
+                    }
+                }
+            },
+            "output": {
+                "type": "integer"
+            }
+        },
+        "GetRaftLog": {
+            "http": {
+                "method": "GET",
+                "requestUri": "/_/metadata/admin/raft_sessions/{LogId}/log?begin={Begin}&end={End}&targetLeader={TargetLeader}"
```

Comment:
should the params `Begin/End/TargetLeader` be required? (since these seem to be hardcoded here)
If they aren't required, I would suggest `"location": "querytstring"` in the member definition instead of "uri".

Reply:
I see, it should be in `querystring` in that case.

Do you have a reference doc for editing this file? The past few times I've had to work with this file have been guess and check (based on previous examples here and in the aws json files), but I haven't been able to dig up any documentation.

Reply:
I don't there are any formal documentation on this. I have been pulling examples from files in https://github.com/aws/aws-sdk-js/tree/master/apis

Reply:
I've been using those examples as well, was hoping for something more clear and definitive.

If it's in the querystring, why does it have to be a required param?

Reply:
Looking at the `aws-sdk` generator code, if the querystrings are defined in `requestUri`, if the substitution values aren't provided, the `requestUri` will be malformed. (e.g. literal string value `{Begin}` if the `Begin` member is missing)

By making these required, the client will throw an error if any these members are missing.

Reply:
My bad, I phrased the question wrong, I was asking if the `location` was `querystring`, why would it be required. Good to confirm that in that case, it isn't. The edits in the initial comment made it seem that the two changes both needed to be made.

Considering other `querystring` params in this file are not specified as required, I will go forward with just changing the `location` param.

On the metadata api side, I am not convinced that all of the parameters are required for the querystring. In the documentation, it specifies a default for `targetLeader` and a default (limited) max return of 10k logs.

Even if the values are required, that would (imo) should still be added during the usage of `BackbeatClient` and not added as a limitation here in the api definitions.

Reply:
You should find the spec and examples in the OpenAPI repo: https://github.com/OAI/OpenAPI-Specification